WorkflowUI/Sources/Screen/AdaptedEnvironmentScreen.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  File.swift
+//
+//
+//  Created by Kyle Van Essen on 3/14/23.
+//
+
+#if canImport(UIKit)
+
+import Foundation
+import ViewEnvironment
+
+/// Wraps a `Screen` tree with a modified `ViewEnvironment`.
+///
+/// By specifying environmental values with this `Screen`, all child screens nested
+/// will automatically inherit those values automatically. Values can be changed
+/// anywhere in a sub-tree by inserting another `AdaptedEnvironmentScreen`.
+///
+/// ```swift
+/// MyScreen(...)
+///     .adaptedEnvironment(keyPath: \.myValue, to: newValue)
+/// ```
+///
+public struct AdaptedEnvironmentScreen<Content: Screen>: Screen {
+    /// The screen wrapped by this screen.
+    public var wrapped: Content
+
+    /// Takes in a mutable `ViewEnvironment` which can be mutated to add or override values.
+    public typealias Adapter = (inout ViewEnvironment) -> Void
+
+    var adapter: Adapter
```

Comment:
Doesn't really matter because this is a value type, but why can't these be `let`? It doesn't look we're using their `var`ness in the implementation, tests, or sample usage.

Reply:
I tend to just keep things as `var` unless `let` is absolutely required (eg, consistency between variables), mostly as a purely stylistic thing.